bb8/src/inner.rs

```diff
@@ -161,6 +161,20 @@ where
         }
     }
 
+    /// Adds an external connection to the pool if there is capacity for it.
+    pub(crate) fn try_put(&self, conn: M::Connection) -> Result<(), M::Connection> {
```

Comment:
It might be better to move this logic into a method on `SharedPool`, which would also allow us to keep `approvals()` private. Then, I think we want a layer here which would run `manager.has_broken()` on the new `conn` before we add it. Might need another error enum to accomodate the different error paths.

Reply:
That makes sense! Let me move things around